### PR TITLE
TLS Improvements

### PR DESCRIPTION
**Pull request by mchindri**
- Update tls for minimal configuration.
- Add support to verify server identity
- Add EINVAL error code  to error.h
- Add support  for secure client identification
- Enable PEM certificate format
- Add memory optimization options

### CODE REVIEW

File: network/tcp_socket.c [hunk 1]
```diff
@@ -164,6 +172,28 @@ static int32_t stcp_socket_init(struct secure_socket_desc **desc,
 					  MBEDTLS_SSL_VERIFY_NONE);
 	}
 
+	if (param->cli_cert) {
+		if (!param->cli_pk) {
```
Comment (mirceacaprioru) [inline network/tcp_socket.c:188]: this can be moved out of this if statement

Comment (mchindri) [reply network/tcp_socket.c:188]: If a user set cli_cert then it should set also cli_pk. On the other side, if cli_cert is not set then the value of cli_pk doesn't matter. I think checking it outside wouldn't make sense in this context.

Comment (mirceacaprioru) [reply network/tcp_socket.c:188]: understood, makes sense.

**Comment by mchindri** [comment]
v2:
 Add missing  #include "noos_mbedtls_config.h" in tcp_socket.c

**Comment by mchindri** [comment]
v3:
 - Add missing endif in aducm.mk

**Pull request merged**